presto-main/src/main/java/com/facebook/presto/server/remotetask/HttpRemoteTaskFactory.java

```diff
@@ -165,4 +181,63 @@ public RemoteTask createRemoteTask(Session session,
                 stats,
                 isBinaryTransportEnabled);
     }
+
+    @Override
+    public void removeRemoteSource(TaskId destinationTaskId, TaskId sourceTaskId, Consumer<PrestoException> onFailure)
+    {
+        // TODO: Check whether we need the onFailure callback. Since this method is used for recoverable execution,
+        //  and if the cleanup failed, the query will fail.
+        URI remoteSourceUri = uriBuilderFrom(locationFactory.createLocalTaskLocation(destinationTaskId))
+                .appendPath("remote-source")
+                .appendPath(sourceTaskId.toString())
+                .build();
+
+        Request request = prepareDelete()
+                .setUri(remoteSourceUri)
+                .build();
+        RequestErrorTracker errorTracker = new RequestErrorTracker(
+                destinationTaskId,
+                remoteSourceUri,
+                maxErrorDuration,
+                errorScheduledExecutor,
+                "Remove exchange remote source");
+
+        executeDestroyExchangeSourceRequest(errorTracker, request, onFailure);
+    }
+
+    // Copied from https://github.com/prestodb/presto/commit/36a60a6f6e029ddd890d0eb10de01f339eedfebb#diff-b040516898118558de47053d230b8420
```

Comment:
Is this comment intentional? It may be better to say `Copied from CLASS::method` or sth like that. Even better & if possible, extract a common utility method to reuse.

Reply:
@nezihyigitbasi : I would remove this comment. I will left a comment in PR to help us know where the code comes from later :)